timeSelectFlag_addition

juchitel1 opened the pull request:
addition of timeSelectFlag that allows the user to manually select clean period for analysis

juchitel1 committed:
timeSelectFlag_addition

addition of timeSelectFlag that allows the user to manually select clean period for analysis

RJCooperUCL commented on LUMO_toolbox/DOTHUB_LUMOdataQualityCheck.m:
```diff
@@ -45,6 +46,11 @@ function DOTHUB_dataQualityCheck(nirsFileName,printFigFlag)
 end
 
+if ~exist('timeSelectFlag','var')
+    timeSelectFlag = 1;
```
I would default to timeSelectFlag = 0.  i.e. if a user doesn't specify it, it is assumed we don't want to use it.

juchitel1 replied:
Changed this

RJCooperUCL commented on LUMO_toolbox/DOTHUB_LUMOdataQualityCheck.m:
```diff
@@ -58,9 +64,91 @@ function DOTHUB_dataQualityCheck(nirsFileName,printFigFlag)
+    my_ans = 'n';
+    while my_ans == 'n';
+    fig1 = figure; set(fig1, 'units', 'normalized', 'outerposition', [0 0 0.8 0.8], 'color', [1 1 1])
+    plot(d(:,1500:2000)); xlim([0 size(d,1)]); ylim([0 1.5]); title('Select segment to include (start-end pair) - Press enter when finished', 'fontsize', 20)
```
Don't hard-code indices - this is something you should always avoid when coding.  What would happen if we were using only 2 tiles?

RJCooperUCL replied:
If you want to reduce the number of channels plotted, try randomly selecting 500 channels, or plotting all channels if number of channels <= 500

juchitel1 replied:
I did this so now it selects 500 channels randomly

RJCooperUCL commented on LUMO_toolbox/DOTHUB_LUMOdataQualityCheck.m:
```diff
@@ -58,9 +64,91 @@ function DOTHUB_dataQualityCheck(nirsFileName,printFigFlag)
+    set(gca, 'Yscale', 'log')
+    hold on
+% do data brushing manually
+    [x, ~] = ginput;
```
Can you add a message to user (command line?) to tell them what they should do (click and drag data selection I guess?)

juchitel1 replied:
Now reads: "%A subset of channels are plotted to decrease computer load (this can
%always be changed to plot all channels). % For manual time period selection, click once at the beginning of the segment to
% select the start point. Drag the cursor and click again to select the end
% point. Hit enter when finished. After, the figure will reappear with the
% selected time period highlighted in blue. This is to make sure that the correct period(s) was(were)
%selected beforre proceeding. If the period is correct, type 'y' and hit enter. "

RJCooperUCL commented on LUMO_toolbox/DOTHUB_LUMOdataQualityCheck.m:
```diff
@@ -58,9 +64,91 @@ function DOTHUB_dataQualityCheck(nirsFileName,printFigFlag)
+%always be changed to plot all channels).
+if (timeSelectFlag==1) 
+    my_ans = 'n';
+    while my_ans == 'n';
```
I don't understand why you need/want a While loop?  Is the idea that people can select a range until they get it right/are happy with it?

juchitel1 replied:
Yeah. When I do this is isn't always clear where the cursor is selecting so having it highlight after helps me make sure that its not getting period with motion

RJCooperUCL commented on LUMO_toolbox/DOTHUB_LUMOdataQualityCheck.m:
```diff
@@ -58,9 +64,91 @@ function DOTHUB_dataQualityCheck(nirsFileName,printFigFlag)
+       clean_idx = ones(size(d,1),1);
+       my_ans = 'y';
+  end
+   data.ini = mrk_data(1,1);
```
Do you need the data variable? Seems unnecessary

juchitel1 replied:
Took this out

juchitel1 committed:
update to time select flag

Update to time select flag to make the default false, also plot a random 500 channels rather than pre-specified channels except if there are less than 500 channels in the rercording. New comments

juchitel1 commented:
Hi, I believe I fixed the above. The changes are under the new commit titled "update to time select flag"
